File: app/src/main/java/com/mytlogos/enterprise/ui/PdfViewerFragment.java
package com.mytlogos.enterprise.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

import com.github.barteksc.pdfviewer.PDFView;
import com.github.barteksc.pdfviewer.scroll.DefaultScrollHandle;
import com.mytlogos.enterprise.R;

import java.io.File;

class PdfViewerFragment extends TextViewerFragment {
    private PDFView pdfDisplay;

    @NonNull
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);
        this.pdfDisplay = view.findViewById(R.id.pdfView);
        this.setHasOptionsMenu(true);
        this.pdfDisplay.setOnClickListener(v -> this.toggleReadingMode());
        return view;
    }

    @Override
    void updateContent() {
        this.pdfDisplay
                .fromFile(new File(this.currentBook))
                .defaultPage(0)
                .onPageScroll((page, positionOffset) -> System.out.printf("Page: %d and Offset: %s%n", page, positionOffset))
                .scrollHandle(new DefaultScrollHandle(this.requireContext()))
                .load();
    }

    @Override
    int getLayoutRes() {
        return R.layout.fragment_reader_pdf;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/ui/PdfViewerFragment.kt
package com.mytlogos.enterprise.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.github.barteksc.pdfviewer.PDFView
import com.github.barteksc.pdfviewer.scroll.DefaultScrollHandle
import com.mytlogos.enterprise.R
import java.io.File

internal class PdfViewerFragment : TextViewerFragment() {
    private lateinit var pdfDisplay: PDFView

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        val view = super.onCreateView(inflater, container, savedInstanceState)
        pdfDisplay = view.findViewById(R.id.pdfView)
        setHasOptionsMenu(true)
        pdfDisplay.setOnClickListener { toggleReadingMode() }
        return view
    }

    override fun updateContent() {
        pdfDisplay
            .fromFile(File(currentBook))
            .defaultPage(0)
            .onPageScroll { page: Int, positionOffset: Float -> println("Page: $page and Offset: $positionOffset") }
            .scrollHandle(DefaultScrollHandle(requireContext()))
            .load()
    }

    override val layoutRes: Int
        get() = R.layout.fragment_reader_pdf
}